web/crux-ui/src/components/products/versions/images/edit-image-json.tsx

```diff
@@ -3,60 +3,115 @@ import useMultiInputState from '@app/components/editor/use-multi-input-state'
 import JsonEditor from '@app/components/shared/json-editor-dynamic-module'
 import { IMAGE_WS_REQUEST_DELAY } from '@app/const'
 import { CANCEL_THROTTLE, useThrottling } from '@app/hooks/use-throttleing'
-import { CompleteContainerConfig, ContainerConfig, InstanceContainerConfig, UniqueKeyValue } from '@app/models'
-import { fold } from '@app/utils'
-import { completeContainerConfigSchema } from '@app/validations'
+import {
+  ContainerConfig,
+  InitContainer,
+  InitContainerVolumeLink,
+  InstanceJsonContainerConfig,
+  JsonContainerConfig,
+  JsonInitContainer,
+  JsonKeyValue,
+  UniqueKey,
+  UniqueKeyValue,
+} from '@app/models'
+import { containerConfigSchema } from '@app/validations'
 import clsx from 'clsx'
 import { CSSProperties, useCallback } from 'react'
 import { v4 as uuid } from 'uuid'
 
-const DEFAULT_CONFIG = completeContainerConfigSchema.getDefault() as any as CompleteContainerConfig
+const DEFAULT_CONFIG = containerConfigSchema.getDefault() as any as ContainerConfig
 
-const keyValueArrayToJson = (envs: UniqueKeyValue[]): Record<string, string> =>
-  fold(envs, {}, (prev, it) => {
-    if (!prev[it.key]) {
-      prev[it.key] = it.value
-    }
+type JsonConfig = InstanceJsonContainerConfig | JsonContainerConfig
+
+const keyValueArrayToJson = (list: UniqueKeyValue[]): JsonKeyValue => {
+  const result = {}
 
-    return prev
+  list?.forEach(it => {
+    result[it.key] = it.value
   })
 
-const imageConfigToCompleteContainerConfig = (
-  currentConfig: CompleteContainerConfig,
-  imageConfig: ContainerConfig | InstanceContainerConfig,
-): CompleteContainerConfig => {
-  if (!imageConfig) {
-    return currentConfig ?? DEFAULT_CONFIG
-  }
+  return result
+}
 
-  const config: CompleteContainerConfig = {
-    ...(imageConfig.config ?? currentConfig ?? DEFAULT_CONFIG),
-    name: imageConfig.name ?? currentConfig.name,
-    environment: currentConfig?.environment ?? {},
-    capabilities: currentConfig?.capabilities ?? {},
-    secrets: currentConfig?.secrets ?? {},
-  }
+const keyArrayToJson = (list: UniqueKey[]): string[] => list?.map(it => it.key)
+
+const simplify = <T,>(item: T): Omit<T, 'id'> => {
+  const newItem: any = { ...item }
+  delete newItem.id
+
+  return newItem
+}
 
-  if (imageConfig.environment) {
-    config.environment = keyValueArrayToJson(imageConfig.environment)
+const imageConfigToJsonContainerConfig = (
```

Comment:
We have to implement this function inside an `edit-image-json` component?

Reply:
Only used in this file.